code/modules/mining/machine_processing.dm

```diff
@@ -156,6 +169,89 @@
 	src.updateUsrDialog()
 	return
 
+/obj/machinery/mineral/processing_unit_console/proc/print_report(var/mob/living/user)
+	if(!inserted_id)
+		to_chat(user, span("warning", "No ID inserted. Cannot digitally sign."))
+		return
+	if(!input_mats.len && !output_mats.len && !alloy_mats)
+		to_chat(user, span("warning", "There is no data to print."))
+		return
+
+	var/obj/item/weapon/paper/P = new /obj/item/weapon/paper(user.loc)
+	var/date_string = worlddate2text()
+	idx++
+
+	var/form_title = "Form 0600 - Mining Yield Declaration"
+	var/dat = "<small><center><b>NanoTrasen Inc.<br>"
+	dat += "Civilian Branch of Operation</b><br><br>"
+
+	dat += "Form 0600<br> Mining Yield Declaration</center><hr>"
+	dat += "Facility: NSS Aurora<br>"
+	dat += "Date: [date_string]<br>"
+	dat += "Index: [idx]<br><br>"
+
+	dat += "Miner(s): <span class=\"paper_field\"></span><br>"
+	dat += "Ore Yields: <br>"
+
+	dat += "<table>"
+	for(var/material/OM in output_mats) {
+		if(output_mats[OM] > 1)
+			dat += "<tr><td><b><small>[output_mats[OM]]</b></small></td><td><small>[OM.display_name] [OM.sheet_plural_name]</small></td></tr>"
+		else
+			dat += "<tr><td><b><small>[output_mats[OM]]</b></small></td><td><small>[OM.display_name] [OM.sheet_singular_name]</small></td></tr>"
+	}
+	for(var/datum/alloy/AM in alloy_mats) {
+		if(alloy_mats[AM] > 1)
+			dat += "<tr><td><b><small>[alloy_mats[AM]]</b></td><td><small>[AM.metaltag] sheets</small></td></tr>"
+		else
+			dat += "<tr><td><b><small>[alloy_mats[AM]]</b></td><td><small>[AM.metaltag] sheet</small></td></tr>"
```

Comment:
```suggestion
			dat += "<tr><td><b><small>[alloy_mats[AM]]</b></td><td><small>[AM.metaltag] sheet</small></td></tr>"
		CHECK_TICK
```

These loops can be very long. So CHECK_TICK is necessary to make sure shit doesn't implode.